### ISSUE TITLE
Optional funnel steps

### ISSUE THREAD

**Issue by lukaszolek** [PostHog/posthog#17313]
## Is your feature request related to a problem?

I have a problem with excluding from my funnels the most important steps (purchases) due to some problems with collecting previous events (e.g. due to ad blocks on the browser).

## Describe the solution you'd like
I'd like to mark some steps as optional - they would be shown as statistics, but lack of their presents would't exclude former events from the funnel.

## Describe alternatives you've considered
Now I need to have separate funnel insights - one for statistical reasons to see how people convert before purchase, and second for the purchase to make sure no purchase event is left.

**Comment by MarconLP** [comment]
+1 https://posthog.com/questions/is-there-a-way-to-make-some-steps-optional-in-a-funnel-as-long-as-conversion-step-is-reached

**Comment by MarconLP** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/10925 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/13387)

**Comment by camerondeleone** [comment]
+1 https://posthog.slack.com/archives/C0368RPHLQH/p1718823983361809

**Comment by camerondeleone** [comment]
+1 https://posthog.slack.com/archives/C07D7PW810R/p1722528045040799

**Comment by tormodfj** [comment]
This would indeed be of great value to my team. We have wizard-like workflows in our apps, where choices you make will affect the path you end up taking towards the end goal, entering or skipping optional steps along the way. 

Being able to represent this in a single funnel would be a major improvement. Optional steps not visited could be marked as skipped for those users/sessions that proceed directly to later steps in the funnel.

This is one of the _few_ things Matomo does well.

**Comment by aleksandra-asanin** [comment]
+1 for optional steps in funnels! there are product flows where some steps can simply be skipped to advance to the next step and this is currently measured as drop-off in posthog but in reality user just skipped an optional step.

**Comment by carter-0** [comment]
+1, I believe this can partially be done using [Actions](https://posthog.com/docs/data/actions) but it would be much better to have a real optional step feature.

**Comment by ordehi** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/31030 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/32639)

**Comment by aspicer** [comment]
@olefredrik Yes, they are still in the works. I have most of the work done but have it currently on the backburner because of a couple more pressing issues that came up. I would look for it to come out next month!
